### PR TITLE
feat: auto-sync — scheduled periodic sync per config

### LINKED ISSUE #12: feat: Auto-Sync — scheduled periodic sync per config
## Overview

Users should not have to manually click Sync on a regular basis. Each config should be able to declare a schedule so the app syncs it automatically.

---

## Schedule Options

Three values only — no custom cron:

| Value | Fires at |
|-------|----------|
| `none` | Never (manual only — current behaviour, default) |
| `weekly` | Every Monday 09:00 JST |
| `monthly` | 1st of every month 09:00 JST |

JST = UTC+9, no DST. All schedule math uses a fixed +9 offset.

---

## Sync Behaviour

When auto-sync fires it runs the exact same logic as the manual Sync button:

1. Wipe all managed blocker events in the config's date range (from today)
2. Recompute freeze days
3. Write new blocker events

No new sync logic — just a new trigger.

The sync runs under the **config owner's OAuth token**, regardless of who enabled the schedule. This mirrors how manual sync works today.

---

## Manual Operations — Disabled When Auto-Sync Is On

When `sync_schedule != none`, the **Sync** and **Wipe** buttons on the config detail page are **greyed out and non-interactive**.

A tooltip on the disabled buttons reads:

> _"Manual operations are disabled while Auto-Sync is on. Disable Auto-Sync first if you want to sync or wipe manually."_

This removes any possibility of a manual/auto race condition without needing locks.

---

## Permissions

- Write and Power users can enable auto-sync on configs they can edit (same rule as editing)
- Power users can enable auto-sync on any config (same rule as editing others' configs)

---

## Missed Windows (Server Downtime)

Auto-sync does **not** catch up after downtime. On server restart, any `next_sync_at` that is in the past is advanced to the next future occurrence before the polling loop starts. Missed windows are silently skipped.

---

## First Sync After Enabling

Enabling a schedule does **not** trigger an immediate sync. The first sync fires at the next natural occurrence of the schedule.

---

### DB Schema (4 new columns on `configs`)

| Column | Type | Default | Notes |
|--------|------|---------|-------|
| `sync_schedule` | TEXT | `'none'` | `none` / `weekly` / `monthly` |
| `next_sync_at` | DATETIME | NULL | UTC; computed when schedule is set |
| `last_auto_synced_at` | DATETIME | NULL | UTC |
| `last_auto_sync_result` | TEXT | NULL | Success message or error string |

`next_sync_at` is precomputed and stored — the poller does a simple `WHERE next_sync_at <= NOW()` query.

### Background Worker

A single goroutine starts at server boot with a 1-minute ticker. Each tick:

1. Query `configs WHERE sync_schedule != 'none' AND next_sync_at <= NOW()`
2. For each result, run sync sequentially (no parallelism needed — one goroutine, one config at a time)
3. Store result in `last_auto_sync_result` and `last_auto_synced_at`
4. Compute and store new `next_sync_at`

Sequential processing by a single goroutine means no locking is needed.

### `nextSyncAt(schedule, from time.Time) time.Time`

Pure helper function — returns the next future scheduled time strictly after `from`. Easy to unit test independently.

### UI Changes

**Config edit form:**
- Add `Auto-Sync` dropdown: `Off` / `Weekly (Mon 09:00 JST)` / `Monthly (1st 09:00 JST)`

**Config detail page:**
- Show schedule badge (e.g. `auto-sync: weekly`)
- Show `Last auto-sync: 2026-05-05 09:00 JST — ✅ 3 events written` or `❌ <error message>`
- Show `Next auto-sync: 2026-05-12 09:00 JST`
- Sync and Wipe buttons: disabled + tooltip when schedule is active

---

## Out of Scope (for now)

- Custom cron expressions
- Sync-on-save after editing YAML
- Failure notifications (email, Slack, webhook)
- Per-replica deduplication (single-replica deployment assumed)

### PR DESCRIPTION
**Pull request by nvatuan**
## Summary

Closes #12.

- **DB**: 4 new columns on `configs` (`sync_schedule`, `next_sync_at`, `last_auto_synced_at`, `last_auto_sync_result`) added via additive `ALTER TABLE` migration so existing databases upgrade without data loss.
- **Scheduler**: `internal/scheduler` package with `NextSyncAt(schedule, from)` pure helper (weekly = every Monday 09:00 JST, monthly = 1st of month 09:00 JST) and a 1-minute ticker worker that runs sequentially in a single goroutine — no locks needed.
- **Missed windows**: On server restart, any `next_sync_at` in the past is silently advanced to the next future occurrence before the ticker starts.
- **Auto-sync runs as config owner**: scheduler uses the config's `user_id` token, matching manual sync behaviour.
- **Manual ops disabled when auto-sync is on**: `HandleSync` and `HandleWipe` return an error response if `sync_schedule != none`; the detail page renders disabled buttons with a descriptive tooltip.
- **UI**: Auto-Sync dropdown on create/edit form; detail page shows schedule badge, last run timestamp + result, and next run time all in JST.

## Test plan

- [ ] `go test ./...` passes (includes new `scheduler_test.go` covering `NextSyncAt` for weekly and monthly edge cases)
- [ ] Create a config with Auto-Sync set to Weekly — verify detail page shows badge and next-run time
- [ ] Verify Sync and Wipe buttons are disabled with tooltip when schedule is active
- [ ] Change schedule to Off — verify buttons re-enable
- [ ] Manually check DB: `sync_schedule`, `next_sync_at` columns populated correctly after save

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/scheduler/scheduler.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+	return &Scheduler{configs: configs, tokens: tokens, oauthCfg: oauthCfg}
+}
+
+// Start runs the scheduler loop. It advances any past-due configs on startup,
+// then processes due configs every minute. Blocks until ctx is cancelled.
+func (s *Scheduler) Start(ctx context.Context) {
+	log := logging.GetLogger()
+	if err := s.advancePastDue(); err != nil {
+		log.WithError(err).Warn("scheduler: failed to advance past-due configs on startup")
+	}
```
Comment (nvatuan) [inline internal/scheduler/scheduler.go]: missing info log sched starts
